src/sqlparser/src/parser.rs

```diff
@@ -1046,11 +1046,35 @@ impl Parser {
         };
 
         if let Some(op) = regular_binary_operator {
-            Ok(Expr::BinaryOp {
-                left: Box::new(expr),
-                op,
-                right: Box::new(self.parse_subexpr(precedence)?),
-            })
+            if let Some(keyword) =
+                self.parse_one_of_keywords(&[Keyword::ANY, Keyword::ALL, Keyword::SOME])
+            {
+                self.expect_token(&Token::LParen)?;
+                let right = self.parse_expr()?;
+                self.expect_token(&Token::RParen)?;
```

Comment:
Can `select 1 < SOME(null);` work?

Reply:
Sorry I mean `select SOME(null) > 1;`

Reply:
No, we can't. `SOME/ANY/ALL` can only appear in the right side of binary op.

Reply:
get it